File: server/scripts/generateRecipes.js
const { faker } = require("@faker-js/faker");

const generateRecipe = () => ({
  category: faker.helpers.arrayElements(
    ["Breakfast", "Lunch", "Dinner", "Snack", "Dessert"],
    3
  ),
  difficulty: faker.helpers.arrayElement(["Easy", "Medium", "Hard"]),
  ingredients: faker.helpers.multiple(
    () => ({
      name: faker.food.ingredient(),
      quantity: faker.number.int({ min: 1, max: 10 }),
    }),
    { count: { min: 3, max: 7 } }
  ),
  name: faker.food.dish(),
  preparationTime: faker.number.int({ min: 10, max: 120 }),
  steps: faker.helpers.multiple(
    () => ({
      description: faker.lorem.sentence(),
    }),
    { count: { min: 1, max: 3 } }
  ),
  visibility: faker.helpers.arrayElement(["Public", "Private"]),
});
const addRecipes = (numRecipes = 5, username) => {
  const recipes = [];
  for (let i = 0; i < numRecipes; i++) {
    const recipe = generateRecipe();
    recipe.username = username;
    recipes.push(recipe);
  }
  return recipes;
};

module.exports = {
  addRecipes,
};

File: server2/scripts/generate_recipes.py
from faker import Faker
import random

fake = Faker()


def generate_recipe():
    return {
        "category": random.sample(
            ["Breakfast", "Lunch", "Dinner", "Snack", "Dessert"],
            k=3
        ),
        "difficulty": random.choice(["Easy", "Medium", "Hard"]),
        "ingredients": [
            {
                "name": fake.word(),
                "quantity": random.randint(1, 10),
                "unit": random.choice(["g", "kg", "ml", "l", "pcs", "tbsp", "tsp"])
            }
            for _ in range(random.randint(3, 7))
        ],
        "name": fake.sentence(nb_words=3).replace(".", ""),
        "preparationTime": random.randint(10, 120),
        "steps": [
            {
                "description": fake.sentence()
            }
            for _ in range(random.randint(1, 3))
        ],
        "visibility": random.choice(["Public", "Private"]),
        "imageUrl": ""
    }


def add_recipes(num_recipes=5, username=None):
    recipes = []

    for _ in range(num_recipes):
        recipe = generate_recipe()
        recipe["username"] = username
        recipes.append(recipe)

    return recipes